lib/Runtime/Library/JavascriptProxy.cpp

```diff
@@ -891,39 +891,49 @@ namespace Js
             // the proxy has been revoked; TypeError.
             if (!threadContext->RecordImplicitException())
                 return FALSE;
-            JavascriptError::ThrowTypeError(GetScriptContext(), JSERR_ErrorOnRevokedProxy, _u("enumerate"));
-        }
-
-        //4. Let trap be the result of GetMethod(handler, "enumerate").
-        //5. ReturnIfAbrupt(trap).
-        //6. If trap is undefined, then
-        //a.Return the result of calling the[[Enumerate]] internal method of target.
-        //7. Let trapResult be the result of calling the[[Call]] internal method of trap with handler as the this value and a new List containing target.
-        //8. ReturnIfAbrupt(trapResult).
-        //9. If Type(trapResult) is not Object, then throw a TypeError exception.
-        //10. Return trapResult.
-        JavascriptFunction* getEnumeratorMethod = GetMethodHelper(PropertyIds::enumerate, scriptContext);
-        Assert(!GetScriptContext()->IsHeapEnumInProgress());
-        if (nullptr == getEnumeratorMethod)
-        {
-            return target->GetEnumerator(enumNonEnumerable, enumerator, requestContext, preferSnapshotSemantics, enumSymbols);
+            JavascriptError::ThrowTypeError(GetScriptContext(), JSERR_ErrorOnRevokedProxy, _u("ownKeys"));
+        }    
+        
+        Var enmeratorObj;
+        Var propertyName = nullptr;
+        PropertyId propertyId;
+        int index = 0;
+        JsUtil::BaseDictionary<const char16*, Var, Recycler> dict(scriptContext->GetRecycler());
+        JavascriptArray* arrResult = scriptContext->GetLibrary()->CreateArray();
+
+        // 13.7.5.15EnumerateObjectProperties(O)#
+        //  for (let key of Reflect.ownKeys(obj)) {    
+        Var trapResult = JavascriptOperators::GetOwnPropertyNames(this, scriptContext);
+        Assert(JavascriptArray::Is(trapResult));
```

Comment:
I would prefer a check in runtime then just Assert, at least a defense in depth.